From b585fdc6092b462454ea4af4ae2a29e863044595 Mon Sep 17 00:00:00 2001
From: Peter Eisenmann <p3732@getgoogleoff.me>
Date: Fri, 12 Jun 2026 21:30:06 +0200
Subject: [PATCH 05/17] general: Propagate natural size when scrollable


diff --git a/src/core/mainWindow.js b/src/core/mainWindow.js
index 217275b483796619be36de372bfb382fadd09d37..b5800ff75d91cf96b99bc3501ca495cdbbb6d3a0 100644
--- a/src/core/mainWindow.js
+++ b/src/core/mainWindow.js
@@ -76,7 +76,6 @@ export class MainWindow extends Adw.ApplicationWindow {
 
     _init(application) {
         this._renderer = null;
-        this._lastWindowSize = [0, 0];
         this.file = null;
 
         this._animating = 0;
diff --git a/src/viewers/evince.js b/src/viewers/evince.js
index 7564331d11c70ee1530b210ec4f3214c8a0b97b7..95d04093609690f24397167ae5dc9745f5239d35 100644
--- a/src/viewers/evince.js
+++ b/src/viewers/evince.js
@@ -12,7 +12,6 @@ import PapersDocument from 'gi://PapersDocument';
 import PapersView from 'gi://PapersView';
 import Sushi from 'gi://Sushi';
 
-import {VIEW_MIN} from '../util/constants.js';
 import {Renderer} from '../core/renderer.js';
 import {ToolbarOverlay} from '../widgets/toolbarOverlay.js';
 
@@ -93,8 +92,8 @@ export const Klass = class PapersRenderer extends ToolbarOverlay {
         this._defineActions();
 
         const scrolledWindow = new Gtk.ScrolledWindow({ visible: true,
-                                                        min_content_height: VIEW_MIN,
-                                                        min_content_width: VIEW_MIN,
+                                                        propagate_natural_height: true,
+                                                        propagate_natural_width: true,
                                                         child: this._view});
         this.set_child(scrolledWindow);
 
diff --git a/src/viewers/libreoffice.js b/src/viewers/libreoffice.js
index 5b62fe22e28226550b282a7cbd7760ec4c62c151..eef1348c782c123f5088a38bb106ed252e11bfbb 100644
--- a/src/viewers/libreoffice.js
+++ b/src/viewers/libreoffice.js
@@ -12,7 +12,6 @@ export const isAvailable = function() {
     return LOKDocView !== undefined;
 };
 
-import {VIEW_MIN} from '../util/constants.js';
 import {Renderer} from '../core/renderer.js';
 
 export const Klass = class LibreofficeRenderer extends Gtk.ScrolledWindow {
@@ -40,9 +39,10 @@ export const Klass = class LibreofficeRenderer extends Gtk.ScrolledWindow {
 
     _init(file) {
         super._init({ hexpand: true,
+                      propagate_natural_height: true,
+                      propagate_natural_width: true,
                       visible: true,
-                      min_content_height: VIEW_MIN,
-                      min_content_width: VIEW_MIN });
+        });
 
         this._lastAllocWidth = 0;
         this._tickCallbackId = 0;
-- 
2.54.0

