From d77063a2e9e802934186852c584912a2390cd540 Mon Sep 17 00:00:00 2001
From: Peter Eisenmann <p3732@getgoogleoff.me>
Date: Fri, 12 Jun 2026 18:13:45 +0200
Subject: [PATCH 07/16] main-window: Cap natural size by max size

---
 src/core/mainWindow.js | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/core/mainWindow.js b/src/core/mainWindow.js
index 277150d..a57b5b9 100644
--- a/src/core/mainWindow.js
+++ b/src/core/mainWindow.js
@@ -213,7 +213,8 @@ export class MainWindow extends Adw.ApplicationWindow {
         if (resizePolicy == ResizePolicy.MAX_SIZE)
             windowSize = maxSize;
         else if (resizePolicy == ResizePolicy.NAT_SIZE)
-            windowSize = natSize;
+            windowSize = [ Math.min(natSize[0], maxSize[0]),
+                           Math.min(natSize[1], maxSize[1]) ];
         else if (resizePolicy == ResizePolicy.SCALED)
             windowSize = _getScaledSize(natSize, maxSize, false);
 
-- 
2.54.0

