From 4f5edbfdd56ab961828d14d6b4c76eb453490e2f Mon Sep 17 00:00:00 2001
From: Peter Eisenmann <p3732@getgoogleoff.me>
Date: Fri, 12 Jun 2026 18:27:07 +0200
Subject: [PATCH 10/16] video: Use scaled resize policy

---
 src/viewers/video.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/viewers/video.js b/src/viewers/video.js
index 58a35bc..11b8a9e 100644
--- a/src/viewers/video.js
+++ b/src/viewers/video.js
@@ -67,7 +67,7 @@ export const Klass = class VideoRenderer extends ToolbarOverlay {
     }
 
     get resizePolicy() {
-        return ResizePolicy.NAT_SIZE;
+        return ResizePolicy.SCALED;
     }
 };
 
-- 
2.54.0

