From 6bf85f4e376212a609aa9be2d07fc956d1608263 Mon Sep 17 00:00:00 2001
From: Peter Eisenmann <p3732@getgoogleoff.me>
Date: Fri, 12 Jun 2026 18:27:07 +0200
Subject: [PATCH 10/17] video: Use scaled resize policy


diff --git a/src/viewers/video.js b/src/viewers/video.js
index 58a35bc569a4cea240cd846c65b67bd377e333aa..11b8a9ef96b344112aec1131985d3654af8ffce9 100644
--- a/src/viewers/video.js
+++ b/src/viewers/video.js
@@ -67,7 +67,7 @@ export const Klass = class VideoRenderer extends ToolbarOverlay {
     }
 
     get resizePolicy() {
-        return ResizePolicy.NAT_SIZE;
+        return ResizePolicy.SCALED;
     }
 };
 
-- 
2.54.0

