From 7c481df6d2ab12d49147c9002af27a6239f53dab Mon Sep 17 00:00:00 2001
From: Peter Eisenmann <p3732@getgoogleoff.me>
Date: Fri, 12 Jun 2026 21:37:55 +0200
Subject: [PATCH 13/17] main-window: Fix error renderer resizing


diff --git a/src/core/mainWindow.js b/src/core/mainWindow.js
index fe3e1a2bdebf45e96e803b662a38b640c9884c2c..42eaa1f39f50d78ebd2a22fc3323f2dc5516e097 100644
--- a/src/core/mainWindow.js
+++ b/src/core/mainWindow.js
@@ -126,7 +126,9 @@ export class MainWindow extends Adw.ApplicationWindow {
             ?? this.file.get_basename()
             ?? this.file.get_uri());
         this.set_title(title);
-        this.set_resizable(this._renderer.resizable);
+        this.set_resizable(renderer.resizable);
+        renderer.connect('notify::ready', this._onRendererReady.bind(this));
+        this._onRendererReady();
     }
 
     _onRendererFullscreen() {
diff --git a/src/viewers/error.js b/src/viewers/error.js
index 611752730b0b42ae9a780a1c1868b30ac1542edb..a87814f1510327a48498c423430e556150bf7451 100644
--- a/src/viewers/error.js
+++ b/src/viewers/error.js
@@ -27,6 +27,10 @@ export class ErrorRenderer extends Adw.Bin {
         }, this);
     }
 
+    get ready() {
+        return !!this._ready;
+    }
+
     _init(error) {
         super._init();
 
-- 
2.54.0

