From a36e970ff5bd965725cc55de23fbd0abdab6ec9f Mon Sep 17 00:00:00 2001
From: FlameFlag <git@ps1.sh>
Date: Sun, 21 Jun 2026 02:30:01 +0300
Subject: [PATCH 14/16] main: Use synchronous GApplication run

---
 src/core/main.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/main.js b/src/core/main.js
index 28f6de0..49fff16 100644
--- a/src/core/main.js
+++ b/src/core/main.js
@@ -35,5 +35,5 @@ export async function main(argv) {
                                         inactivity_timeout: 12000 });
     if (GLib.getenv('SUSHI_PERSIST'))
         application.hold();
-    await application.runAsync(argv);
+    application.run(argv);
 }
-- 
2.54.0

